models/shopcart.py

```diff
@@ -17,11 +17,25 @@ class Shopcart(object):
 
     __products = dict()
 
-    def __init__(self, uid=0, products=None):
+    def __init__(self, uid=0, products=[]):
```

Comment:
You should never have a list (immutable object) as a default argument in python cause it has some funky default behavior: http://docs.python-guide.org/en/latest/writing/gotchas/.

A better alternative is leave `None` as the default argument and then in the body do something like `self.products = products if products else []`.

That's a one-liner, but you can also break it out into a regular if/else statement.

Reply:
Nice to know. Thanks @thoughts1053